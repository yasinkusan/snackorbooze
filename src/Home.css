.Home-info {
	color: #9e4a36;
	text-decoration: none;
	border-bottom: 2px solid transparent;
	transition: all .5s ease-in-out;
}

.Home-info:hover {
	text-decoration: none;
	color: #cf8369;
	border-bottom: 2px solid #9e4a36;
}
