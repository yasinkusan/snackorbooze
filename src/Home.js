import React from 'react';
import { Link } from 'react-router-dom';
import './Home.css';
import { Card, CardBody, CardTitle, CardText } from 'reactstrap';

function Home({ snacks, drinks }) {
	return (
		<section className="Home">
			<Card>
				<CardBody className="text-center">
					<CardTitle>
						<h3 className="font-weight-bold">Welcome to Silicon Valley's premier dive cafe!</h3>
					</CardTitle>
					<CardText>
						{/* display # of snacks and drinks; will change as snacks/drinks are added/removed */}
						<h3>
							We have{' '}
							<Link className="Home-info" to="/snacks">
								{snacks.length} snacks
							</Link>{' '}
							and {' '}
							<Link className="Home-info" to="/drinks">
								{drinks.length} drinks
							</Link>{' '}
							to choose from!
						</h3>
					</CardText>
				</CardBody>
			</Card>
		</section>
	);
}

export default Home;
