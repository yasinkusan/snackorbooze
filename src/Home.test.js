import React from 'react';
import Home from './Home';
import { MemoryRouter } from 'react-router-dom';

import { render, fireEvent } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

function sleep(time) {
	return new Promise((resolve) => setTimeout(resolve, time));
}
const snacks = [ "i'm a snack" ];
const drinks = [ "i'm a drink" ];

it('renders without crashing', () => {
	render(
		<MemoryRouter initialEntries={[ '/' ]}>
			<Home snacks={snacks} drinks={drinks} />
		</MemoryRouter>
	);
});

it('matches loading snapshot', () => {
	const { asFragment } = render(
		<MemoryRouter initialEntries={[ '/' ]}>
			<Home snacks={snacks} drinks={drinks} />
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
it('matches rendered snapshot', async () => {
	const { asFragment } = render(
		<MemoryRouter initialEntries={[ '/' ]}>
			<Home snacks={snacks} drinks={drinks} />
		</MemoryRouter>
	);
	await act(() => sleep(1100));
	expect(asFragment()).toMatchSnapshot();
});
