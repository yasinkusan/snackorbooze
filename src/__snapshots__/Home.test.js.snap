// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches loading snapshot 1`] = `
<DocumentFragment>
  <section
    class="Home"
  >
    <div
      class="card"
    >
      <div
        class="text-center card-body"
      >
        <h5
          class="card-title"
        />
        <h3
          class="font-weight-bold"
        >
          Welcome to Silicon Valley's premier dive cafe!
        </h3>
        <p
          class="card-text"
        />
        <h3>
          We have 
          <a
            class="Home-info"
            href="/snacks"
          >
            1 snacks
          </a>
           and  
          <a
            class="Home-info"
            href="/drinks"
          >
            1 drinks
          </a>
           to choose from!
        </h3>
        <p />
      </div>
    </div>
  </section>
</DocumentFragment>
`;

exports[`matches rendered snapshot 1`] = `
<DocumentFragment>
  <section
    class="Home"
  >
    <div
      class="card"
    >
      <div
        class="text-center card-body"
      >
        <h5
          class="card-title"
        />
        <h3
          class="font-weight-bold"
        >
          Welcome to Silicon Valley's premier dive cafe!
        </h3>
        <p
          class="card-text"
        />
        <h3>
          We have 
          <a
            class="Home-info"
            href="/snacks"
          >
            1 snacks
          </a>
           and  
          <a
            class="Home-info"
            href="/drinks"
          >
            1 drinks
          </a>
           to choose from!
        </h3>
        <p />
      </div>
    </div>
  </section>
</DocumentFragment>
`;
